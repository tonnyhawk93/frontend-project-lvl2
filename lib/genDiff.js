import { createFileObject, compareObjects } from './lib.js';
import { stylish } from './formaters/index.js';

export default (filepath1, filepath2, formater = stylish) => {
  const obj1 = createFileObject(filepath1);
  const obj2 = createFileObject(filepath2);
  const diff = compareObjects(obj1, obj2);
  return formater(diff);
};
