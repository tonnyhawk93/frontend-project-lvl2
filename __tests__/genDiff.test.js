import { fileURLToPath } from 'url';
import path from 'path';
import genDiff from '../lib/genDiff';
import { plain, json } from '../lib/formaters/index.js';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const getFixturePath = (filename) => path.join(__dirname, '..', '__fixtures__', filename);

const yamlFilepath1 = getFixturePath('file1.yaml');
const yamlFilepath2 = getFixturePath('file2.yaml');
const ymlFilepath1 = getFixturePath('file1.yml');
const ymlFilepath2 = getFixturePath('file2.yml');
const jsonComplicatedFile1 = getFixturePath('complicatedFile1.json');
const jsonComplicatedFile2 = getFixturePath('complicatedFile2.json');

describe('genDiff', () => {
  it('Возращает правильный diff у разных yaml файлов', () => {
    expect(genDiff(yamlFilepath1, yamlFilepath2)).toMatchSnapshot();
  });
  it('Возращает правильный diff у одинаковых yaml файлов', () => {
    expect(genDiff(yamlFilepath1, yamlFilepath1)).toMatchSnapshot();
  });
  it('Возращает правильный diff у разных yml файлов', () => {
    expect(genDiff(ymlFilepath1, ymlFilepath2)).toMatchSnapshot();
  });
  it('Возращает правильный diff у одинаковых yml файлов', () => {
    expect(genDiff(ymlFilepath1, ymlFilepath1)).toMatchSnapshot();
  });
  it('Возращает правильный diff у сложных json файлов', () => {
    expect(genDiff(jsonComplicatedFile1, jsonComplicatedFile2)).toMatchSnapshot();
  });
  it('Правильно работает плоское форматирование у сложных json файлов', () => {
    expect(genDiff(jsonComplicatedFile1, jsonComplicatedFile2, plain)).toMatchSnapshot();
  });
  it('Правильно работает json форматер у файлов', () => {
    expect(genDiff(jsonComplicatedFile1, jsonComplicatedFile2, json)).toMatchSnapshot();
  });
});
