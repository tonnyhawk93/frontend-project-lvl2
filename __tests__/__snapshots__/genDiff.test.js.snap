// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genDiff Возращает правильный diff у одинаковых json файлов 1`] = `
"{
    follow: false
    host: hexlet.io
    proxy: 123.234.53.22
    timeout: 50
}"
`;

exports[`genDiff Возращает правильный diff у одинаковых yaml файлов 1`] = `
"{
    common: {
        setting1: Value 1
        setting2: 200
        setting3: true
        setting6: {
            doge: {
                wow: 
            }
            key: value
        }
    }
    group1: {
        baz: bas
        foo: bar
        nest: {
            key: value
        }
    }
    group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
}"
`;

exports[`genDiff Возращает правильный diff у одинаковых yml файлов 1`] = `
"{
    follow: false
    host: hexlet.io
    proxy: 123.234.53.22
    timeout: 50
}"
`;

exports[`genDiff Возращает правильный diff у разных json файлов 1`] = `
"{
  - follow: false
    host: hexlet.io
  - proxy: 123.234.53.22
  - timeout: 50
  + timeout: 20
  + verbose: true
}"
`;

exports[`genDiff Возращает правильный diff у разных yaml файлов 1`] = `
"{
    common: {
      + follow: false
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: null
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
              - wow: 
              + wow: so much
            }
            key: value
          + ops: vops
        }
    }
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
  + group3: {
        deep: {
            id: {
                number: 45
            }
        }
        fee: 100500
    }
}"
`;

exports[`genDiff Возращает правильный diff у разных yml файлов 1`] = `
"{
  - follow: false
    host: hexlet.io
  - proxy: 123.234.53.22
  - timeout: 50
  + timeout: 20
  + verbose: true
}"
`;

exports[`genDiff Возращает правильный diff у сложных json файлов 1`] = `
"{
    common: {
      + follow: false
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: null
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
              - wow: 
              + wow: so much
            }
            key: value
          + ops: vops
        }
    }
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
  + group3: {
        deep: {
            id: {
                number: 45
            }
        }
        fee: 100500
    }
}"
`;

exports[`genDiff Правильно работает json форматер у файлов 1`] = `"{\\"common\\":[{\\"value\\":{\\"follow\\":[{\\"value\\":false,\\"type\\":\\"+\\"}],\\"setting1\\":[{\\"value\\":\\"Value 1\\",\\"type\\":\\" \\"}],\\"setting2\\":[{\\"value\\":200,\\"type\\":\\"-\\"}],\\"setting3\\":[{\\"value\\":true,\\"type\\":\\"-\\"},{\\"value\\":null,\\"type\\":\\"+\\"}],\\"setting4\\":[{\\"value\\":\\"blah blah\\",\\"type\\":\\"+\\"}],\\"setting5\\":[{\\"value\\":{\\"key5\\":\\"value5\\"},\\"type\\":\\"+\\"}],\\"setting6\\":[{\\"value\\":{\\"doge\\":[{\\"value\\":{\\"wow\\":[{\\"value\\":\\"\\",\\"type\\":\\"-\\"},{\\"value\\":\\"so much\\",\\"type\\":\\"+\\"}]},\\"type\\":\\" \\"}],\\"key\\":[{\\"value\\":\\"value\\",\\"type\\":\\" \\"}],\\"ops\\":[{\\"value\\":\\"vops\\",\\"type\\":\\"+\\"}]},\\"type\\":\\" \\"}]},\\"type\\":\\" \\"}],\\"group1\\":[{\\"value\\":{\\"baz\\":[{\\"value\\":\\"bas\\",\\"type\\":\\"-\\"},{\\"value\\":\\"bars\\",\\"type\\":\\"+\\"}],\\"foo\\":[{\\"value\\":\\"bar\\",\\"type\\":\\" \\"}],\\"nest\\":[{\\"value\\":{\\"key\\":\\"value\\"},\\"type\\":\\"-\\"},{\\"value\\":\\"str\\",\\"type\\":\\"+\\"}]},\\"type\\":\\" \\"}],\\"group2\\":[{\\"value\\":{\\"abc\\":12345,\\"deep\\":{\\"id\\":45}},\\"type\\":\\"-\\"}],\\"group3\\":[{\\"value\\":{\\"deep\\":{\\"id\\":{\\"number\\":45}},\\"fee\\":100500},\\"type\\":\\"+\\"}]}"`;

exports[`genDiff Правильно работает плоское форматирование у сложных json файлов 1`] = `
"Property 'common.follow' was added with value: false
Property 'common.setting2' was removed
Property 'common.setting3' was updated. From true to null
Property 'common.setting4' was added with value: 'blah blah'
Property 'common.setting5' was added with value: [complex value]
Property 'common.setting6.doge.wow' was updated. From '' to 'so much'
Property 'common.setting6.ops' was added with value: 'vops'
Property 'group1.baz' was updated. From 'bas' to 'bars'
Property 'group1.nest' was updated. From [complex value] to 'str'
Property 'group2' was removed
Property 'group3' was added with value: [complex value]"
`;
